jax/_src/scipy/stats/bernoulli.py

```diff
@@ -42,23 +42,30 @@ def cdf(k: ArrayLike, p: ArrayLike) -> Array:
   k, p = jnp._promote_args_inexact('bernoulli.cdf', k, p)
   zero, one, one_minus_p = _lax_const(k, 0), _lax_const(k, 1), _lax_const(k, 1-p)
   conds = [
-    jnp.isnan(k),
     lax.lt(k, zero),
     jnp.logical_and(lax.ge(k, zero), lax.lt(k, one)),
     lax.ge(k, one)
     ]
-  vals = [jnp.nan, zero, one_minus_p, one]
+  vals = [zero, one_minus_p, one]
   return jnp.where(
-    jnp.logical_and(lax.ge(p, zero), lax.le(p, one)),
-    jnp.select(conds, vals), jnp.nan
+    jnp.logical_or(
+      jnp.logical_or(jnp.isnan(k), jnp.isnan(p)),
+      jnp.logical_or(lax.lt(p, zero), lax.gt(p, one))),
+    jnp.nan, 
+    jnp.select(conds, vals)
   )
 
 @_wraps(osp_stats.bernoulli.ppf, update_doc=False)
 def ppf(q: ArrayLike, p: ArrayLike) -> Array:
   q, p = jnp._promote_args_inexact('bernoulli.ppf', q, p)
-  zero, one = _lax_const(q, 0), _lax_const(q, 1)
-  return jnp.where(
-    jnp.logical_or(jnp.isnan(q), jnp.isnan(p)),
+  zero, one, one_minus_p = _lax_const(q, 0), _lax_const(q, 1), _lax_const(q, 1-p)
+  return jnp.where(jnp.logical_or(
+      jnp.logical_or(jnp.isnan(q), jnp.isnan(p)),
+      jnp.logical_or(
+        jnp.logical_or(lax.lt(p, zero), lax.gt(p, one)),
+        jnp.logical_or(lax.lt(q, zero), lax.gt(q, one))
+        ),
+    ),
```

Comment:
Here and above, I think it would be much more readable to use operators like `<`, `>`, and `|`. So instead of
```python
jnp.logical_or(
      jnp.logical_or(jnp.isnan(q), jnp.isnan(p)),
      jnp.logical_or(
        jnp.logical_or(lax.lt(p, zero), lax.gt(p, one)),
        jnp.logical_or(lax.lt(q, zero), lax.gt(q, one))
        ),
    )
```
you could write
```python
jnp.isnan(q) | jnp.isnan(p) | (p < 0) | (p > 1) | (q < 0) | (q > 1)
```
This approach is slightly slower at trace time, but is much easier to read and will end up lowering to the same low-level XLA operations.